virtcontainers/sandbox.go

```diff
@@ -1163,6 +1170,26 @@ func (s *Sandbox) StopContainer(containerID string) (VCContainer, error) {
 		return nil, err
 	}
 
+	// Check if hypervisor is running, if not, skip the following steps
```

Comment:
Is it better to do hypervisor process existed checking here: https://github.com/kata-containers/runtime/blob/f4da3f585a4b1af54226c52bc1c816fb06aa6bb1/virtcontainers/kata_agent.go#L1420 as below: 
`
if _, err := k.sendReq(req); err != nil {
		if running, _ := utils.IsProcRunning(sandbox.hypervisor.pid()); !running {
			k.Logger().WithField("sandbox", sandbox.id).WithField("container", c.id).Warning("Failed to stop container, but vm not exist")
		} else {
			return err
		}
	}
`
thus there is no need to do the cleanup here.

Reply:
@Ace-Tang Please follow the approach proposed by @lifupan above. It will help not to add any extra cleanup code.

Reply:
There are some cleanup like unmount rootfs to do, so I choose to put this logic in stop, which stopContainer and stopSanbox all call stop function